pkg/networkservice/chains/nsmgr/suite_test.go

```diff
@@ -451,53 +470,145 @@ func (s *nsmgrSuite) Test_PassThroughRemoteUsecase() {
 	defer cancel()
 
+	nsReg := linearNS(nsesCount)
+	nsReg, err := s.domain.Nodes[0].NSRegistryClient.Register(ctx, nsReg)
+	require.NoError(t, err)
+
 	var nsesReg [nsesCount]*registry.NetworkServiceEndpoint
 	for i := 0; i < nsesCount; i++ {
+		nsesReg[i] = &registry.NetworkServiceEndpoint{
+			Name:                fmt.Sprintf("endpoint-%v", i),
+			NetworkServiceNames: []string{nsReg.GetName()},
+			NetworkServiceLabels: map[string]*registry.NetworkServiceLabels{
+				nsReg.GetName(): {
+					Labels: map[string]string{
+						step: fmt.Sprintf("%v", i),
+					},
+				},
+			},
+		}
+
 		var additionalFunctionality []networkservice.NetworkServiceServer
-		if i != 0 {
+		if i != nsesCount-1 {
 			additionalFunctionality = []networkservice.NetworkServiceServer{
 				chain.NewNetworkServiceServer(
 					clienturl.NewServer(s.domain.Nodes[0].NSMgr.URL),
 					connect.NewServer(ctx,
-						sandbox.NewCrossConnectClientFactory(sandbox.GenerateTestToken,
-							newPassTroughClient(fmt.Sprintf("my-service-remote-%v", i-1)),
-							kernel.NewClient()),
+						client.NewClientFactory(
+							client.WithName(fmt.Sprintf("%v", i)),
+							client.WithAdditionalFunctionality(
+								mechanismtranslation.NewClient(),
+								injectlabels.NewClient(nsesReg[i].NetworkServiceLabels[nsReg.Name].Labels),
+								kernel.NewClient(),
+							),
+						),
 						connect.WithDialTimeout(sandbox.DialTimeout),
 						connect.WithDialOptions(sandbox.DefaultDialOptions(sandbox.GenerateTestToken)...),
 					),
 				),
 			}
 		}
 
-		nsReg, err := s.domain.Nodes[0].NSRegistryClient.Register(ctx, &registry.NetworkService{
-			Name: fmt.Sprintf("my-service-remote-%v", i),
-		})
+		_, err = s.domain.Nodes[0].NewEndpoint(ctx, nsesReg[i], sandbox.GenerateTestToken, additionalFunctionality...)
 		require.NoError(t, err)
+	}
+
+	nsc := s.domain.Nodes[0].NewClient(ctx, sandbox.GenerateTestToken)
+
+	request := defaultRequest(nsReg.Name)
+
+	conn, err := nsc.Request(ctx, request)
+	require.NoError(t, err)
+	require.NotNil(t, conn)
+
+	// Path length to first endpoint is 5
+	// Path length from NSE client to other local endpoint is 5
+	require.Equal(t, 5*(nsesCount-1)+5, len(conn.Path.PathSegments))
+
+	// Refresh
+	conn, err = nsc.Request(ctx, request)
+	require.NoError(t, err)
+
+	// Close
+	_, err = nsc.Close(ctx, conn)
+	require.NoError(t, err)
+
+	// Endpoint unregister
+	for i := 0; i < len(nsesReg); i++ {
+		_, err = s.domain.Nodes[0].EndpointRegistryClient.Unregister(ctx, nsesReg[i])
+		require.NoError(t, err)
+	}
+}
 
+func (s *nsmgrSuite) Test_PassThroughLocalUsecaseMultiLabel() {
+	t := s.T()
+	const nsesCount = 9
+
+	labels := multiLabelEndpoints()
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Hour)
+	defer cancel()
+
+	nsReg := multiLabelNS()
+	nsReg, err := s.domain.Nodes[0].NSRegistryClient.Register(ctx, nsReg)
+	require.NoError(t, err)
+
+	var nsesReg [nsesCount]*registry.NetworkServiceEndpoint
+	for i := 0; i < nsesCount; i++ {
```

Comment:
We can use here such loop instead of this:
```go
var labelA, labelB string
for i := 0; i < 3; i++ {
    labelA += "a"
    for k := 0; k < 3; k++ {
        labelB += "b"
        labels := map[string]string{
            "label_a": labelA,
            "label_b": labelB,
        }
        ...
        if i != 2 || k != 2 { // let expect any "aaa" or "bbb" endpoint not to be passthrough
            ...
        }
    }
}
```
So we don't actually need either `nsesCount` or `multiLabelEndpoints()`.

Reply:
ok, done